fmralign/metrics.py

```diff
@@ -0,0 +1,180 @@
+    In: Chung A., Gee J., Yushkevich P., and Bao S. (eds) Information 
+    Processing in Medical Imaging. Lecture Notes in Computer Science,
+    vol 11492. Springer, Cham.
+    DOI: 10.1007/978-3-030-20351-1_18.`
+    """
+    if y_true.ndim == 1:
+        y_true = y_true.reshape((-1, 1))
+
+    if y_pred.ndim == 1:
+        y_pred = y_pred.reshape((-1, 1))
+
+    numerator = ((y_true - y_pred) ** 2).sum(axis=0, dtype=np.float64)
+    denominator = ((y_true) ** 2).sum(axis=0, dtype=np.float64)
+
+    # Include only non-zero values
+    nonzero_denominator = (denominator != 0)
+    nonzero_numerator = (numerator != 0)
+    valid_score = (nonzero_denominator & nonzero_numerator)
+
+    # Calculate reconstruction error
+    output_scores = np.ones([y_true.shape[-1]])
+    output_scores[valid_score] = 1 - (numerator[valid_score] /
+                                      denominator[valid_score])
+    if multioutput == 'raw_values':
+        # return scores individually
+        return output_scores
+    
+    elif multioutput == 'uniform_average':
+        # passing None as weights yields uniform average
+        return np.average(output_scores, weights=None)
+
+    elif multioutput == 'variance_weighted':
+        # if providing sample_weights directly
+        if sample_weights is not None:
+            weight = sample_weights[:, np.newaxis]
+        else:
+            weight = 1.
+        avg_weights = (weight * (y_true - np.average(
+            y_true, axis=0, weights=sample_weights)) ** 2).sum(axis=0,
+                                                               dtype=np.float64)
+        # avoid failing on constant y or one-element arrays
+        if not np.any(nonzero_denominator):
+            if not np.any(nonzero_numerator):
+                return 1.0
+            else:
+                return 0.0
+        return np.average(output_scores, weights=avg_weights)
+
+
+def reconstruction_ratio(aligned_error, identity_error):
+    """
+    Calculates the reconstruction error
+    as defined by Bazeille and
+    colleagues (2019).
+    
+    A value greater than 0 indicates that
+    voxels are predicted better by aligned data
+    than by raw data.
+    
+    Parameters
+    ----------
+    aligned_error : float64
+        The reconstruction error from a given
+        functional alignment method
+    identity error :  float64
+        The reconstruction error from predicting
+        the target subject as the source subject
+
+    References
+    ----------
+    `Bazeille T., Richard H., Janati H., and Thirion B. (2019) Local
+    Optimal Transport for Functional Brain Template Estimation.
+    In: Chung A., Gee J., Yushkevich P., and Bao S. (eds) Information 
+    Processing in Medical Imaging. Lecture Notes in Computer Science,
+    vol 11492. Springer, Cham.
+    DOI: 10.1007/978-3-030-20351-1_18.`
+    """
+    num = 1 - aligned_error
+    den = 1 - identity_error
+    return 1 - (num / den)
```

Comment:
no need to handle den=0 ?

Reply:
Should be handled now :smile_cat: Right now I'm returning 0 if den is 0, but let me know if there's another solution you'd prefer !